ggu/main.c

```diff
@@ -68,25 +76,17 @@ process_single_policy(const struct ggu_policy *policy, const struct ggu_config *
 	entry->u.ggu.state = policy->state;
 	rte_memcpy(&entry->u.ggu.flow, &policy->flow, sizeof(entry->u.ggu.flow));
 
-	switch(policy->state) {
+	switch (policy->state) {
 	case GK_GRANTED:
-		entry->u.ggu.params.u.granted.tx_rate_kb_sec =
-			rte_be_to_cpu_32(
-			policy->params.u.granted.tx_rate_kb_sec);
-		entry->u.ggu.params.u.granted.cap_expire_sec =
-			rte_be_to_cpu_32(
-			policy->params.u.granted.cap_expire_sec);
-		entry->u.ggu.params.u.granted.next_renewal_ms =
-			rte_be_to_cpu_32(
-			policy->params.u.granted.next_renewal_ms);
-		entry->u.ggu.params.u.granted.renewal_step_ms =
-			rte_be_to_cpu_32(
-			policy->params.u.granted.renewal_step_ms);
+		rte_memcpy(&entry->u.ggu.params.u.granted,
+			&policy->params.u.granted,
+			sizeof(entry->u.ggu.params.u.granted));
```

Comment:
To make things typesafe here, we should use `entry->u.ggu.params.u.granted = policy->params.u.granted;` here.